src/analysisd/analysisd.c

```diff
@@ -1848,7 +1848,7 @@ void * w_writer_log_thread(__attribute__((unused)) void * args ){
     #ifdef PRELUDE_OUTPUT_ENABLED
                 /* Log to prelude */
                 if (Config.prelude) {
-                    if (Config.prelude_log_level <= currently_rule->level) {
+                    if (Config.prelude_log_level <= lf->generated_rule->level) {
```

Comment:
We should also check if `lf->generated_rule` is not `NULL` in order to avoid a possible segmentation fault.

Reply:
Of course!